src/hifz/dataserver.py

```diff
@@ -0,0 +1,29 @@
+import csv
+from pathlib import Path
+
+from hifz.models import Card
+
+
+class DataServer:
+    def __init__(self, file_name: str) -> None:
+        self.file_name = file_name
+        self._cards = self.__read_file()  # Read the file during initialization
+
+    def __read_file(self) -> list[Card]:
+        """Private helper method to read the file and store the data."""
+        cards = []
+
+        try:
+            with Path(self.file_name).open() as file:
+                csv_reader = csv.reader(file)
+                for row in csv_reader:
+                    cards.append(Card(*row))
```

Comment:
What happens if we switch the order of front and back in the csv? This requires the ordering of row match the constructor of the Card class.

Reply:
I think we can use a DictReader to make it more generalizable and readable.